pkgs/development/compilers/llvm/15/libcxxabi/default.nix

```diff
@@ -68,6 +68,13 @@ stdenv.mkDerivation rec {
   ] ++ lib.optionals (stdenv.hostPlatform.useLLVM or false) [
     "-DLLVM_ENABLE_LIBCXX=ON"
     "-DLIBCXXABI_USE_LLVM_UNWINDER=ON"
+  ] ++ lib.optionals ((stdenv.hostPlatform.useLLVM or false) ||
+                      (stdenv.hostPlatform.isDarwin && enableShared)) [
+    # libcxxabi's CMake looks as though it treats -nostdlib++ as implying -nostdlib,
+    # but that does not appear to be the case for example when building
+    # pkgsLLVM.libcxxabi (which uses clangNoCompilerRtWithLibc).
+    "-DCMAKE_EXE_LINKER_FLAGS=-nostdlib"
+    "-DCMAKE_SHARED_LINKER_FLAGS=-nostdlib"
```

Comment:
What is the reason for this? I would think we *do* wanna link libc but *don't* want to link (an already-existing) C++ standard library implementation. `-nostdlib++` without `-nostdlib` therefore seems correct to me.

Reply:
Apologies, this is coming back to it some time after I implemented it, so I have forgotten the detail here and I have limited cycles to spare to dive back in. I suspect it has to do with making cmake configure, or linking succeed. I presume if you take it out you'll find the same failure I experienced? If not, maybe there is something further downstream which is broken.

Reply:
OK thanks @pwaller. So far, it actually seems to be working fine for me.

Reply:
@pwaller: I think it's (1); see: https://github.com/NixOS/nixpkgs/pull/311836/commits/c84a96bd96ae719a6f9b679e84424c86dbf4b6cf from #311836

Reply:
Thanks for the cross reference. I looked into it a little, I suspect the issue this was trying to fix went away when libcxxabi was merged into libcxx.